File: Day 1: Arrays/Stock Buy and Sell.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int min=prices[0],maxi=0;
        for(int i=1;i<prices.size();i++){
            if(prices[i]<min) min=prices[i];
            if(prices[i]>min) maxi=max(maxi,prices[i]-min);
        }
        return maxi;
    }
};

File: Day 1: Arrays/Java/Stock Buy and Sell.java
class Solution {
    public int maxProfit(int[] prices) {
        int res = 0;
        int mini = prices[0];
        for(int i=1;i<prices.length;i++){
            res = Math.max(res,prices[i]-mini);
            mini = Math.min(mini,prices[i]);
        }
        return res;
    }
}

//O(N) time and O(1) space
